pkg/util/strategicpatch/diff.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategicpatch
+
+import (
+	"io"
+	"strings"
+)
+
+type diffWriter struct {
+	Writer io.Writer
```

Comment:
Yup. Instead of defining the struct like:
```go
type diffWriter struct {
	Writer io.Writer
	Depth  int
}
```
try:
```go
type diffWriter struct {
	io.Writer
	Depth  int
}
```
then you can call Write() directly on *diffWriter which simplifies your code below and allows you to remove the delegated `func (dw *diffWriter) Write(out string)` method.